fabfile/tasks/helpers.py

```diff
@@ -806,6 +806,38 @@ def setup_hugepages_node(*args):
 def setup_hugepages():
     setup_hugepages_node(env.host_string)
 
+@task
+def setup_coremask_node(*args):
+    """Setup core mask on one or list of nodes
+    USAGE: fab setup_coremask_node:user@host1,user@host2,...
+    """
+    vrouter_file = '/etc/contrail/supervisord_vrouter_files/contrail-vrouter-dpdk.ini'
+    taskset_param = ''
+
+    for host_string in args:
+        dpdk = getattr(env, 'dpdk', None)
+        if dpdk:
+            # do not crash when coremask was not specified in testbed
+            try:
+                coremask = dpdk[env.host_string]['coremask']
+            except KeyError:
+                return
+        else:
+            return
+
+        if (coremask == ''):
+            return
+
+        # supported coremask format: hex: (0x3f); list: (0,3-5), (0,1,2,3,4,5)
+        # if specified as a list of cpus, -c flag must be provided to taskset
+        if ',' in coremask or '-' in coremask:
+            taskset_param = ' -c'
+
+        # prepend taskset only if valid coremask was specified in testbed
+        with settings(host_string=host_string):
+            sudo('sed -i \'s/command=/command=taskset%s %s /\' %s' \
+                %(taskset_param, coremask, vrouter_file))
```

Comment:
>What happens if the core mask is not well formed
Actually there is not thorough validation, so coremask like "asdf" is accepted and taskset will fail. I assumed this option will be used by someone who knows what they're doing. Would you like to get proper validation?

>can we make the provisioning fail?
Yes, it is possible. Would you like make coremask a mandatory option? Currently, when no coremask is specified (or it is empty), it is simply ignored and taskset does not get prepended to the command line in the .ini file. It is totally harmless to the setup, as vRouter will take care of starting without coremask provided (4 cores by default). Would you like to change that behavior?

>can the core mask be specified in the same place as the huge pages
Could you be more specific? It is in the same script, in the separate function, just below the huge page function.

>I don;t think we should require a separate fab command for this.
There is no separate command. It all gets executed during fab setup_all. If I do not understand you correctly, will you please elaborate more on your requirements?